src/loki_config.py

```diff
@@ -97,7 +97,7 @@ def config(self, coordinator: Coordinator) -> str:
             ),
             "memberlist": self._memberlist_config(
                 cluster_label=f"{coordinator._charm.model.name}-cluster",
-                worker_addresses=coordinator.cluster.gather_addresses(),
+                worker_addresses=set(coordinator.cluster.gather_addresses()),
```

Comment:
It is weird that we need to transform a tuple into a set, since we are already doing the opposite in the [method's definition](https://github.com/canonical/cos-lib/blob/main/src/cosl/coordinated_workers/interface.py#L346)

If we need a set, shouldn't `gather_addresses` return a set instead of a tuple?

Reply:
We're returning a Tuple because if we don't return an ordered collection, the different address ordering will cause the worker to restart unnecessarily ([more info](https://github.com/canonical/cos-lib/pull/86))